artemis-server/src/test/resources/ConfigurationTest-full-config.xml

```diff
@@ -406,6 +406,9 @@
       <page-max-concurrent-io>17</page-max-concurrent-io>
       <read-whole-page>true</read-whole-page>
       <journal-directory>somedir2</journal-directory>
+      <journal-history-directory>history</journal-history-directory>
```

Comment:
this config looks neat, but I think "journal-retention" is a better name, and I think in high volume scenarios second granularity will be necessary to capture use cases less than days.

Reply:
@gtully I thought about using max-files for such cases.. Humans are usually slow to react... ..

perhaps we could have this

<journal-retention period-type="days|files|hours" period="Number" folder="data/history"/>

Reply:
files is good if you care about you max disk usage. But I think for time, hours is too coarse. and seconds covers everything up to hours nicely.

Reply:
period does not work for files: maybe:

<journal-retention limit-type="files|days|hours|seconds" limit="number > 0" folder="data/history"/>

Reply:
<journal-retention  limit-type="files|days|hours|seconds" limit="number > 0" folder="data/history"/>

Reply:
What if you want to limit by files and time at same time.

Reply:
I would suggest a timebased config item and a files/size based config item

Reply:
yep, I agree, time and space, probably need both, to limit disk usage in case the time limits is too large.

Reply:
``

Reply:
> files is good if you care about you max disk usage. But I think for time, hours is too coarse. and seconds covers everything up to hours nicely.

Initial idea was for it to be coarse as it wood be days/weeks, intent is to time machine back after say an incident that gets found out after the fact and work out what could have happened. I’m not sure how useful seconds here would be tbh.

Reply:
@michaelandrepearce some time ago, I talked to an Architect at my job who wanted to get diffs from the broker and send over to another site from time to time. 

This kind of feature would be useful for such case maybe?


It wouldn't hurt to have it though.. it's just an additional option.


I am thinking to have a commented out xml on the broker.xml that will have 7 days, 10G as the config. So if users just uncomment that they will have the 7 days and 10G max disk.

Reply:
Hmmm.. this feature could also be used for Disaster and Recovery options.

Reply:
@michaelandrepearce / @gtully actually.. I take that back...


I will do days and hours...


I don't have time to test all the options now.. having more code means I would have to test it...


If we ever cross a use case of a user needing more granularity we can add it later.

Reply:
I was going to take it back, but I actually wrote a test to validate the options on the XML.